sdks/java/extensions/zetasketch/src/test/java/org/apache/beam/sdk/extensions/zetasketch/HllCountTest.java

```diff
@@ -0,0 +1,357 @@
+import org.apache.beam.sdk.testing.NeedsRunner;
+import org.apache.beam.sdk.testing.PAssert;
+import org.apache.beam.sdk.testing.TestPipeline;
+import org.apache.beam.sdk.transforms.Create;
+import org.apache.beam.sdk.util.CoderUtils;
+import org.apache.beam.sdk.values.KV;
+import org.apache.beam.sdk.values.PCollection;
+import org.apache.beam.sdk.values.TypeDescriptor;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.rules.ExpectedException;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+/** Tests for {@link HllCount}. */
+@RunWith(JUnit4.class)
+public class HllCountTest {
+
+  @Rule public final transient TestPipeline p = TestPipeline.create();
+  @Rule public transient ExpectedException thrown = ExpectedException.none();
+
+  // Integer
+  private static final List<Integer> INTS1 = Arrays.asList(1, 2, 3, 3, 1, 4);
+  private static final byte[] INTS1_SKETCH;
+  private static final Long INTS1_ESTIMATE;
+
+  static {
+    HyperLogLogPlusPlus<Integer> hll = new HyperLogLogPlusPlus.Builder().buildForIntegers();
+    INTS1.forEach(hll::add);
+    INTS1_SKETCH = hll.serializeToByteArray();
+    INTS1_ESTIMATE = hll.longResult();
+  }
+
+  private static final List<Integer> INTS2 = Arrays.asList(2, 3, 4, 3, 2, 6, 5);
+  private static final byte[] INTS2_SKETCH;
+  private static final Long INTS2_ESTIMATE;
+
+  static {
+    HyperLogLogPlusPlus<Integer> hll = new HyperLogLogPlusPlus.Builder().buildForIntegers();
+    INTS2.forEach(hll::add);
+    INTS2_SKETCH = hll.serializeToByteArray();
+    INTS2_ESTIMATE = hll.longResult();
+  }
+
+  private static final byte[] INTS1_INTS2_SKETCH;
+
+  static {
+    HyperLogLogPlusPlus<?> hll = HyperLogLogPlusPlus.forProto(INTS1_SKETCH);
+    hll.merge(INTS2_SKETCH);
+    INTS1_INTS2_SKETCH = hll.serializeToByteArray();
+  }
+
+  // Long
+  private static final List<Long> LONGS = Arrays.asList(1L, 2L, 3L, 3L, 1L, 4L);
+  private static final byte[] LONGS_SKETCH;
+
+  static {
+    HyperLogLogPlusPlus<Long> hll = new HyperLogLogPlusPlus.Builder().buildForLongs();
+    LONGS.forEach(hll::add);
+    LONGS_SKETCH = hll.serializeToByteArray();
+  }
+
+  // String
+  private static final List<String> STRINGS = Arrays.asList("s1", "s2", "s3", "s3", "s1", "s4");
+  private static final byte[] STRINGS_SKETCH;
+
+  static {
+    HyperLogLogPlusPlus<String> hll = new HyperLogLogPlusPlus.Builder().buildForStrings();
+    STRINGS.forEach(hll::add);
+    STRINGS_SKETCH = hll.serializeToByteArray();
+  }
+
+  private static final int TEST_PRECISION = 20;
+  private static final byte[] STRINGS_SKETCH_TEST_PRECISION;
+
+  static {
+    HyperLogLogPlusPlus<String> hll =
+        new HyperLogLogPlusPlus.Builder().normalPrecision(TEST_PRECISION).buildForStrings();
+    STRINGS.forEach(hll::add);
+    STRINGS_SKETCH_TEST_PRECISION = hll.serializeToByteArray();
+  }
+
+  // Bytes
+  private static final List<byte[]> BYTES;
```

Comment:
Alternatively to the code below, enumerate the bytes here and cast them to byte, e.g., 
Arrays.asList((byte) 0xa, (byte) 0x4, ...); -- that might be easier to read and you don't need to handle any exceptions.

Reply:
Done. (Just that we need a List<byte[]> instead of List<byte>)